### PR TITLE
docs: add zap executor

### PR DESCRIPTION
**Pull request by vLia**
## Pull request description 

Docs for ZAP executor

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]

File: docs/docs/test-types/executor-zap.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
+import Admonition from "@theme/Admonition";
+
+# OWASP Zed Attack Proxy executor
```
Comment (jfermi) [inline docs/docs/test-types/executor-zap.md:3]:
```suggestion
# OWASP Zed Attack Proxy Executor
```

File: docs/docs/test-types/executor-zap.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+Starting from version 1.12, Testkube has a dedicated executor for running ZAP tests. All you need to do is populate a file with the necessary parameters, and create a Testkube test.
```
Comment (jfermi) [inline docs/docs/test-types/executor-zap.md:5]:
```suggestion
Starting from version 1.12, Testkube has a dedicated executor for running ZAP tests. All you need to do is populate a file with the necessary parameters and create a Testkube test.
```

File: docs/docs/test-types/executor-zap.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+          <li>It is free and open-source, and also the most widely used web app scanner in the world. It provides a high range of options for security automation.</li>
+        </ul>
+      </Admonition>
+    </div>
+  );
+}
+
+## **Creating a ZAP Test**
+
+The official ZAP Docker image, on which the executor was built, lets you run three types of tests: baseline, full and api scans. Depending on which of these functionalities you want to leverage, the test creation looks slightly different. Not only the type of the test (`--type`) needs to be specified differently, but also the configuration file will have some parameters that do not work with all types.
```
Comment (jfermi) [inline docs/docs/test-types/executor-zap.md:33]:
```suggestion
The official ZAP Docker image, on which the executor was built, lets you run three types of tests: baseline, full and API scans. Depending on which of these functionalities you want to leverage, the test creation looks slightly different. Not only the type of the test (`--type`) needs to be specified differently, but the configuration file will also have some parameters that do not work with all types.
```

File: docs/docs/test-types/executor-zap.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+10017	WARN	(Cross-Domain JavaScript Source File Inclusion - Passive/release)
+10019	WARN	(Content-Type Header Missing - Passive/release)
+10020	WARN	(X-Frame-Options Header Scanner - Passive/release)
+10021	WARN	(X-Content-Type-Options Header Missing - Passive/release)
+.
+.
+.
+```
+
+For further files that should be passed in as parameter to the executor, use the `--copy-files` feature when developing locally. These files will be uploaded to the path `/data/uploads` by default.
```
Comment (jfermi) [inline docs/docs/test-types/executor-zap.md:126]:
```suggestion
For additional files that should be passed in as parameter to the executor, use the `--copy-files` feature when developing locally. These files will be uploaded to the path `/data/uploads` by default.
```

File: docs/docs/test-types/executor-zap.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+testkube get tests
+```
+
+To get a more detailed report, you can run:
+
+```bash
+testkube get execution zap-api-test-1
+```
+
+Another indicator of how the tests went are the report files created. These are uploaded as artifacts into the object storage used by Testkube, and can be retrieved using both the Testkube CLI and the Testkube UI.
```
Comment (jfermi) [inline docs/docs/test-types/executor-zap.md:181]:
```suggestion
Another indicator of how the tests completed are the report files created. These are uploaded as artifacts into the object storage used by Testkube, and can be retrieved using both the Testkube CLI and the Testkube UI.
```

**Review by jfermi — COMMENTED** [review]
A few editing suggestions.

**Pull request merged**